### PR TITLE
W-3 show available localizations

### PR DESCRIPTION
**Pull request by truwer81**

### CODE REVIEW

File: src/main/java/com/example/server/Server.java [hunk 1]
```diff
@@ -7,28 +7,37 @@
...
+import com.example.server.localization.GetLocalizationsService;
+import com.example.server.localization.GetLocalizationsRepository;
 
-import java.net.http.HttpClient;
 import java.util.Objects;
 
 public class Server {
 
     private final LocalizationController localizationController;
+    //private final GetLocalizationsController getLocalizationsController;
```
Comment (MisQl) [inline src/main/java/com/example/server/Server.java]: unnecessary code - to be removed

File: src/main/java/com/example/server/Server.java [hunk 2]
```diff
@@ -7,28 +7,37 @@
...
+        GetLocalizationsRepository getLocalizationsRepository = new GetLocalizationsRepository(sessionFactory);
+        GetLocalizationsService getLocalizationsService = new GetLocalizationsService(getLocalizationsRepository);
+        LocalizationController localizationController = new LocalizationController(objectMapper, localizationService, getLocalizationsService);
         this.localizationController = localizationController;
+
     }
 
     // mapujemy requesty HTTP na metody kontrolera
-    public String callServer(String method, String path, String json) {
+    public String callServer(String method, String path, String json) throws JsonProcessingException {
```
Comment (MisQl) [inline src/main/java/com/example/server/Server.java:28]: JsonProcessingException is caught in the LocalizationController

File: src/main/java/com/example/server/localization/GetLocalizationsRepository.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+public class GetLocalizationsRepository {
+
+    private final SessionFactory sessionFactory;
+
+    public GetLocalizationsRepository(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    public List<Localization> findAll() {
```
Comment (MisQl) [inline src/main/java/com/example/server/localization/GetLocalizationsRepository.java]: can be moved to the LocalizationRepository

File: src/main/java/com/example/server/localization/GetLocalizationsRepository.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        Session session = sessionFactory.openSession();
+        try {
```
Comment (MisQl) [inline src/main/java/com/example/server/localization/GetLocalizationsRepository.java]: start trasaction

File: src/main/java/com/example/server/localization/GetLocalizationsRepository.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            // Utworzenie zapytania HQL do pobrania wszystkich lokalizacji
+            String hql = "FROM Localization";
+            Query<Localization> query = session.createQuery(hql, Localization.class);
+
+            // Wykonanie zapytania i pobranie wyników
+            List<Localization> localizations = query.getResultList();
+            return localizations;
+        } finally {
+
```
Comment (MisQl) [inline src/main/java/com/example/server/localization/GetLocalizationsRepository.java]: unnecessary empty line

File: src/main/java/com/example/server/localization/GetLocalizationsService.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.List;
+
+public class GetLocalizationsService {
+    private final GetLocalizationsRepository getLocalizationsRepository;
+
+    public GetLocalizationsService(GetLocalizationsRepository getLocalizationsRepository) {
+        this.getLocalizationsRepository = getLocalizationsRepository;
+    }
+
+    public List<Localization> getAllLocalizations() {
```
Comment (MisQl) [inline src/main/java/com/example/server/localization/GetLocalizationsService.java]: can be moved to the LocalizationService

File: src/main/java/com/example/server/localization/LocalizationController.java [hunk 7]
```diff
@@ -3,15 +3,20 @@
...
 
+import java.util.List;
+import java.util.stream.Collectors;
+
 // @RequiredArgsConstructor - zamiast konstuktora
 public class LocalizationController {
 
     private final ObjectMapper objectMapper;
     private final LocalizationService localizationService;
+private final GetLocalizationsService getLocalizationsService;
```
Comment (MisQl) [inline src/main/java/com/example/server/localization/LocalizationController.java]: format your code

File: src/main/java/com/example/server/localization/LocalizationController.java [hunk 8]
```diff
@@ -32,6 +37,20 @@ public String createLocalization(String json) {
             return "{\"error\": \"Internal server error\"}"; // http 500
         }
     }
+    // GET /localizations
+    public String getLocalizations() throws JsonProcessingException {
+        try{
+            List<Localization> localizations = getLocalizationsService.getAllLocalizations();
+            return objectMapper.writeValueAsString(localizations
```
Comment (MisQl) [inline src/main/java/com/example/server/localization/LocalizationController.java]: try this line `objectMapper.writeValueAsString(localizations);

**Review by MisQl — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/server/Server.java (hunk 2 above)
Comment (truwer81) [reply src/main/java/com/example/server/Server.java:28]: In IntelliJ, the method localizationController.getLocalizations() is not permitted to execute without handling the exception com.fasterxml.jackson.core.JsonProcessingException. It recommends including this exception in the method's signature to manage potential errors gracefully.

File: src/main/java/com/example/server/localization/LocalizationController.java (hunk 8 above)
Comment (truwer81) [reply src/main/java/com/example/server/localization/LocalizationController.java]: super, zadziałało

**Pull request merged**